### PR TITLE
all: clean up lint issues, remove more dead code

### PR DESCRIPTION
**Pull request by fjl**
The "delete dead code" commit fixes https://github.com/ethereum/go-ethereum/issues/2377

### CODE REVIEW

File: vendor/github.com/Azure/azure-sdk-for-go/storage/blob.go [hunk 1]
```diff
@@ -685,8 +685,8 @@ func (b BlobStorageClient) ReleaseLease(container string, name string, currentLe
 
 	_, err := b.leaseCommonPut(container, name, headers, http.StatusOK)
 	if err != nil {
-		return err
-	}
+	return err
+}
```
Comment (karalabe) [inline vendor/github.com/Azure/azure-sdk-for-go/storage/blob.go]: Please revert this format change.

Comment (fjl) [inline vendor/github.com/Azure/azure-sdk-for-go/storage/blob.go]: It's already reverted ;)

Comment (karalabe) [inline vendor/github.com/Azure/azure-sdk-for-go/storage/blob.go]: Reverted yes, but the formatting got broken in the revertal.

Comment (fjl) [inline vendor/github.com/Azure/azure-sdk-for-go/storage/blob.go]: done

**Comment by ferhatelmas** [comment]
@fjl @karalabe should we enable gosimple on CI. Right now I checked, there is no reported issue so very small diff.

**Comment by fjl** [comment]
@ferhatelmas yes please! `gosimple` is awesome!

**Comment by ferhatelmas** [comment]
@fjl #15593 thanks :smile:

**Pull request merged**